src/action.js

```diff
@@ -29,39 +36,35 @@ async function run() {
 
     const token = core.getInput("github-token");
     const octokit = github.getOctokit(token);
-    const data = await octokit.graphql(
-      `
-      query getLinkedIssues($owner: String!, $name: String!, $number: Int!) {
-        repository(owner: $owner, name: $name) {
-          pullRequest(number: $number) {
-            closingIssuesReferences {
-              totalCount
-            }
-          }
-        }
-      }
-      `,
-      {
-        owner: owner.login,
-        name,
-        number,
-      }
-    );
+    const data = await getLinkedIssues(octokit, name, number, owner.login);
 
     core.debug(`
     *** GRAPHQL DATA ***
     ${format(data)}
     `);
 
-    const linkedIssuesCount =
-      data?.repository?.pullRequest?.closingIssuesReferences?.totalCount;
+    const pullRequest = data?.repository?.pullRequest;
+    const linkedIssuesCount = pullRequest?.closingIssuesReferences?.totalCount;
 
     core.setOutput("linked_issues_count", linkedIssuesCount);
 
     if (!linkedIssuesCount) {
-      core.setFailed(
-        `No linked issues found. Please add the corresponding issues in the pull request description.`
-      );
+      const subjectId = pullRequest?.id;
+
+      if (subjectId) {
+        await addComment(octokit, subjectId);
+        core.debug(`Comment added for ${subjectId} PR`);
+      }
+
+      core.setFailed(ERROR_MESSAGE);
+    } else {
+      // getting only github-actions comment ids
+      const nodeIds = pullRequest?.comments?.nodes
```

Comment:
I think we should have another query for this one, instead of retrieving the first 100 comments on the PR on the previous query... what if there are more comments?

Reply:
Yes, I think the same, but at the same time I'm not sure if it's worth doing a pagination query to get them all and then filter them to get only the ones we need to exclude.

The arguments we pass to get comments are pretty much related to pagination (https://docs.github.com/en/github-ae@latest/graphql/reference/objects#pullrequest)

what do you think?

Reply:
Can't we fire a query with an argument in order to get the only comment we are searching for?

Reply:
![Screen Shot 2022-01-13 at 12 05 54](https://user-images.githubusercontent.com/1309417/149355460-a0a020c4-6e0e-4415-aaa8-dea705c389b0.png)

After dive into the documentation I found just this option to get all comments from a specific PR.
There is no way to search for a specific comments

Reply:
Ouch, yeah I see... I was also searching for that… and I found this action: https://github.com/peter-evans/create-or-update-comment 🤔  What if we use it instead of building this functionality within our action? Would make sense?

Reply:
I don't know if it can be easily integrated. Otherwise we can follow a similar approach to retrieve the comment we need... but I guess that it's basically what we're already doing 😅

Reply:
hmm I'm not sure, I have some questions but I need to dive into that.
How we would add conditional to add or not comments as it is a new workflow?
Also, how about to delete/edit those previous comments? we would face the same issue I guess.

Reply:
exactly!

Reply:
> I don't know if it can be easily integrated. Otherwise we can follow a similar approach to retrieve the comment we need... but I guess that it's basically what we're already doing 😅

exactly! haha

Reply:
Maybe take a look at https://octokit.github.io/rest.js/v18#pagination because seems that is easiest to auto paginate through the PR comments until you find the one you need!

Reply:
Great, yeah probably the best option we have, going to implement that.
Thanks Sr.